### PR TITLE
Add support for "App" mappings without a Bundle class

### LINKED ISSUE #15: Bundle 'App' does not exist.
Staring from Symfony 4, bundle conception has been rejected and is no longer used so how can I create mappings without it?  

* Symfony 5.1
* ElasticSearch 7.1
* This bundle 5.2.6.6

**ongr_elasticsearch.yaml**
```yaml
ongr_elasticsearch:
    managers:
        default:
            index:
                index_name: book
                hosts:
                    - '%env(ELASTICSEARCH_URL)%'
            mappings:
                - App
```
**services.yaml**
```yaml
services:
    _defaults:
        autowire: true
        autoconfigure: true
        bind:
            $manager: '@es.manager'
```
**BookController.php**
```php
class BookController extends AbstractController
{
    /**
     * @Route("/book_popularity", methods={"GET"})
     */
    public function getPopularBooks(Request $request, Manager $manager)
    {
        $repo = $manager->getRepository(Book::class);
        $search = $repo->createSearch();

        $query = new TermQuery('title', 'Harry Potter');
        $search->addQuery($query);

        $result = $repo->findDocuments($search);

        return new JsonResponse($result);
    }
}
```
**Book.php**
```php
/**
 * @ES\Document(type="ms_card_event")
 */
class Book
{
    /**
     * @ES\Property(type="text")
     */
    protected string $title;
    // constructor, getter & setterr
}
```

Without mapping I don't getting any results.

### PR DESCRIPTION
**Pull request by AndreasA**
It is not optimal because it "fakes" a bundle by using a class in the app's source root - by default `App\Kernel` or optionally any other class. The advantage is that in most cases it works out of the box and in some others it can be fixed using a minor configuration change.

The main advantage is, that as there is no bundle configured there will be no unintended side consequences (as I mentioned in the corresponding issue).

Fixes: #15

### CODE REVIEW

File: DependencyInjection/Configuration.php [hunk 1]
```diff
@@ -47,6 +47,7 @@ public function getConfigTreeBuilder()
                     'If profiler is disabled the tracer service will be disabled as well.'
                 )
             ->end()
+            ->scalarNode('app_root_class')->defaultValue('App\\Kernel')->end()
```
Comment (alexander-schranz) [inline DependencyInjection/Configuration.php]: maybe we should set this parameter only when there exists no `App\AppBundle` class. 

```suggestion
            ->scalarNode('app_root_class')->defaultValue(class_exists('App\\AppBundle') ? null : 'App\\Kernel')->end()
```

File: Mapping/DocumentFinder.php [hunk 2]
```diff
@@ -30,11 +32,18 @@ class DocumentFinder
      * Constructor.
      *
      * @param array $bundles Parameter kernel.bundles from service container.
+     * @param string $appRootClass A class that has to be in the app's root folder. By default, the app's kernel class is used.
+     *                             This is to ensure the fake bundle "App" can be used for mappings.
+     *                             It is not a perfect solution but the easiest.
      */
-    public function __construct(array $bundles)
+    public function __construct(array $bundles, string $appRootClass = 'App\\Kernel')
```
Comment (alexander-schranz) [inline Mapping/DocumentFinder.php]:
```suggestion
    public function __construct(array $bundles, ?string $appRootClass = null)
```

File: Mapping/DocumentFinder.php [hunk 3]
```diff
@@ -30,11 +32,18 @@ class DocumentFinder
...
     {
         $this->documentDir = 'Document';
         $this->bundles = $bundles;
+
+        if (!empty($appRootClass) && class_exists($appRootClass)) {
```
Comment (alexander-schranz) [inline Mapping/DocumentFinder.php]:
```suggestion
        if (!$appRootClass && class_exists($appRootClass)) {
```

File: Resources/doc/mapping.md [hunk 4]
```diff
@@ -3,6 +3,21 @@
 Elasticsearch bundle requires mapping definitions for it to work with complex operations,
 like insert and update documents, do a full-text search, etc.
 
+### App mapping
+
+In order for the app's classes to be mapped a fake bundle with the name `App` is added. This is not a perfect solution but was the easiest to implement and add support for it.
+
+It requires a class in the project's source root folder.
+
+By default, the `App\kernel` class is used. This class can be changed to any other class by using the configuration:
```
Comment (alexander-schranz) [inline Resources/doc/mapping.md]:
```suggestion
By default, the `App\Kernel` class is used. This class can be changed to any other class by using the configuration:
```

File: Resources/doc/mapping.md [hunk 5]
```diff
@@ -3,6 +3,21 @@
...
+
+```yaml
+ongr_elasticsearch:
+  app_root_class: 'App\FakeAppRoot'
```
Comment (alexander-schranz) [inline Resources/doc/mapping.md]:
```suggestion
    app_root_class: 'App\YourKernel'
```

**Review by alexander-schranz — CHANGES_REQUESTED** [review]

File: Resources/doc/mapping.md [hunk 6]
```diff
@@ -257,7 +272,7 @@ class Product
     /**
      * @var ContentMetaObject
      *
-     * @ES\Embedded(class="AppBundle:CategoryObject")
+     * @ES\Embedded(class="App\Dcouemnt\CategoryObject")
```
Comment (alexander-schranz) [inline Resources/doc/mapping.md]:
```suggestion
     * @ES\Embedded(class="App\Document\CategoryObject")
```

**Review by alexander-schranz — APPROVED** [review]

File: Mapping/DocumentFinder.php [hunk 7]
```diff
@@ -30,11 +30,18 @@ class DocumentFinder
      * Constructor.
      *
      * @param array $bundles Parameter kernel.bundles from service container.
+     * @param string $appRootClass A class that has to be in the app's root folder. By default, the app's kernel class is used.
```
Comment (alexander-schranz) [inline Mapping/DocumentFinder.php]: looks like the CS does not like something going over 120 characters.

https://github.com/handcraftedinthealps/ElasticsearchBundle/actions/runs/3336329010/jobs/5521759726

Comment (AndreasA) [reply Mapping/DocumentFinder.php]: weird that it only is an issue for the 7.4 pipeline thought 😄

**Comment by alexander-schranz** [comment]
@AndreasA Thank you!

**Pull request merged**